### PR TITLE
Feat/lesq 11/downloads layout [LESQ-11]

### PR DESCRIPTION
**Pull request by njblore**
## Description

- Deleted old download card
- Renamed DLCard to `ResourceCard` so it can be used for share
- made a new component `ResourcePageLayout` to handle the layout for download & share pages
- added some logic to organise the resouce cards according to ACS
- moved some components from the `LessonDownloads.page` into `ResourcePageLayout`

## How to test

I've added a few test cases in storybook on the `DownloadCardGroup` component to cover the different scenarios of resources present (with & without presentation, with one/two worksheets) : https://deploy-preview-2064--owa-storybook.netlify.thenational.academy/?path=/story/download-components-download-card-group--all-resources 

1. Go to https://deploy-preview-2064--oak-web-application.netlify.thenational.academy/teachers/programmes/combined-science-secondary-ks4-foundation-ocr/units/measuring-waves/lessons/transverse-waves/downloads?preselected=all
3. You should see the new layout
4. Go to https://deploy-preview-2064--oak-web-application.netlify.thenational.academy/teachers/programmes/religious-education-secondary-ks4-l/units/islamic-practices-690d/lessons/shahadah-6gu68c/downloads?preselected=all
5. See fewer resources

## Screenshots

How it used to look (delete if n/a):
Screenshot 2023-11-01 at 09 20 45

How it should now look:

Screenshot 2023-11-01 at 09 30 02

## ACs

- [ ]  Column x 2 format (resources on left and details on right)
- [ ]  Swap to new resource selector component
    - Slide deck appears on its own row
    - Unless there is only 1 worksheet resource, in which case that buddies up with the slide deck on the top row
    - Everything else pairs with its no answer/answer variant
    - Everything just moves up if there is no slide deck
- [ ]  Remove old download card component
- [ ]  Reposition t’s & c’s, copyright details and download button
- [ ]  Heading changes - check figma notes

### CODE REVIEW

File: src/components/DownloadComponents/DownloadCardGroup/DownloadCardGroup.tsx [hunk 1]
```diff
@@ -1,92 +1,78 @@
...
   control: Control<DownloadFormProps>;
   hasError?: boolean;
-  errorMessage?: string;
-  onSelectAllClick: () => void;
-  onDeselectAllClick: () => void;
-  preselectAll: boolean;
   triggerForm: () => void;
 };
 
+const DownloadCardGrid = styled(Box)`
```
Comment (JBR90) [inline src/components/DownloadComponents/DownloadCardGroup/DownloadCardGroup.tsx:22]: Could this be styled(Grid) - or maybe could the gird-template-areas functionally be added it the Grid component ? - Maybe it's a bit too custom for a generic grip component though

**Review by JBR90 — APPROVED** [review]
Looks great, the re-usable layout is going to save a lot of time with the share page.
One question around grid but not a blocker

File: src/components/DownloadComponents/DownloadCardGroup/DownloadCardGroup.tsx (hunk 1 above)
Comment (njblore) [reply src/components/DownloadComponents/DownloadCardGroup/DownloadCardGroup.tsx]: good shout! I updated the Grid component to accept templates and used the Grid component directly

**Review by jack-skerrett-oak — APPROVED** [review]

**Pull request merged**